internal/testutils/xds/e2e/server.go

```diff
@@ -120,10 +121,10 @@ type ManagementServerOptions struct {
 // StartManagementServer initializes a management server which implements the
 // AggregatedDiscoveryService endpoint. The management server is initialized
 // with no resources. Tests should call the Update() method to change the
-// resource snapshot held by the management server, as required by the test
-// logic. When the test is done, it should call the Stop() method to cleanup
-// resources allocated by the management server.
-func StartManagementServer(opts ManagementServerOptions) (*ManagementServer, error) {
+// resource snapshot held by the management server, as per by the test logic.
+//
+// Registers a cleanup function on t to stop the management server.
+func StartManagementServer(t *testing.T, opts ManagementServerOptions) *ManagementServer {
```

Comment:
Not at all required as part of this change. But just a thought. 

Should we add `t` as a field to ManagementServer struct?  That way we can call t.Funcs() from the other methods. For eg: Update could call into T.Fatalf() directly?

Reply:
Calling `t.Fatal` from anywhere other than the main test goroutine or test helpers is not an ideal thing to do for the following reasons:
- when the test fails, we are deep down in some function which does not have all the context available at the main test goroutine. This means that the error message that is emitted cannot capture everything that is useful for debugging.
- line number printed in the test output would be that of some function deep down the stack and not from the main test goroutine.

Reply:
Noticed that there was no call to `t.Helper` here, so added that. Thanks.